src/plonkish/backend/halo2.rs

```diff
@@ -258,14 +264,51 @@ impl<F: PrimeField + From<u64> + Hash> ChiquitoHalo2<F> {
 
         for lookup in self.plonkish_ir.lookups.iter() {
             let annotation: &'static str = Box::leak(lookup.annotation.clone().into_boxed_str());
-            meta.lookup_any(annotation, |meta| {
-                let mut exprs = Vec::new();
-                for (src, dest) in lookup.exprs.iter() {
-                    exprs.push((self.convert_poly(meta, src), self.convert_poly(meta, dest)))
+
+            let mut single_fixed_queries = vec![];
+            let mut other_queries = vec![];
+            for (idx, (_, dest)) in lookup.exprs.iter().enumerate() {
+                match dest {
+                    PolyExpr::Query((column, _, _), _) => {
+                        if column.ctype == cFixed
```

Comment:
@leolara here I added the explicit fixed column type check but this may be unnecessary if we know for sure that lookups only involve fixed columns.

Reply:
yes, but this seems more flexible